### ISSUE TITLE
extra least_squares_solvers require documentation

### ISSUE THREAD

**Issue by arvoelke** [nengo/nengo#1065]
If I pass `solver = nengo.utils.least_squares_solvers.RandomizedSVD()` to a connection, I get the following validation error:

```
...
c:\users\aaron\ctn\nengo\nengo\connection.pyc in __init__(self, pre, post, synapse, function, transform, solver, learning_rule_type, eval_points, scale_eval_points, label, seed, modulatory)
    343         self.eval_points = eval_points  # Must be set before function
    344         self.function_info = function  # Must be set after transform
--> 345         self.solver = solver  # Must be set before learning rule
    346         self.learning_rule_type = learning_rule_type  # set after transform
    347         self.modulatory = modulatory
...
ValidationError: Connection.solver: '<nengo.utils.least_squares_solvers.RandomizedSVD object at 0x00000000B6BB5C18>' is not a solver
```

**Comment by jgosmann** [comment]
[Look here](https://github.com/nengo/nengo/pull/803#issuecomment-175258863), I suppose `randomized_svd` was renamed to `RandomizedSVD`?

And I suppose this really needs better documentation.

**Comment by arvoelke** [comment]
Oh geez. Ok thank you!

**Comment by hunse** [comment]
Yeah, that's why they're off in their own file now. The things in `nengo.utils.least_squares_solvers` aren't full solvers, they're subsolvers that a Solver object can use for actually solving the system. That's why they don't subclass Solver.

**Comment by arvoelke** [comment]
Thanks. By the way it works really well! Solved a 20,000 x 2,500 matrix quite quickly and with ideal accuracy, without any tweaking!

**Comment by hunse** [comment]
I think documenting could be as easy as putting a sentence in the docstring, saying that these work with the solvers in `solvers.py` and to look there.

**Comment by tbekolay** [comment]
Could add this to #1017, tracks the documentation stuff I didn't get to in 2.1.0.

**Comment by arvoelke** [comment]
Closed via #1305.
